README.md

```diff
@@ -1,10 +1,140 @@
 # ndx-nirs Extension for NWB
 
+**WIP**
+
+_ideas to include_:
+* purpose of the extension
+  * bring NIRS into nwb
+  * interoperability with tools, etc
+* modes of nirs that it is compatible with
+* based off of SNIRF
+  * expect to release conversion tool
+* main architecture
+
+
 ## Installation
 
+To install after cloning the extension repo from github, execute the following from the root of the repo:
+
+```
+$ pip install .
+```
+
+For development purposes, it might be useful to install in editable mode:
+
+```
+$ pip install -e .
+```
+
+Note: a version of the ndx-nirs package will be released on PyPI before the extension is submitted to the NDX Catalog.
+
 
 ## Usage
 
+```python
+from datetime import datetime
+
+import numpy as np
+
+from pynwb import NWBHDF5IO
+from pynwb.file import NWBFile, Subject
+from hdmf.common import DynamicTableRegion
+
+from ndx_nirs import NIRSSourcesTable, NIRSDetectorsTable, NIRSChannelsTable, NIRSDevice, NIRSSeries
+
+
+# create a basic NWB file
+nwb = NWBFile(
+    session_description="A NIRS test session",
+    identifier="nirs_test_001",
+    session_start_time=datetime.now().astimezone(),
+    subject=Subject(subject_id="nirs_subj_01")
+)
+
+
+# create and populate a NIRSSourcesTable containing the
+# label and location of optical sources for the device
+sources = NIRSSourcesTable()
+# add a row for each source
+sources.add_row(label="S1", x=-1.0, y=0.0, z=0.0)
+sources.add_row(label="S2", x=1.0, y=0.0, z=0.0)
+
+
+# create and populate a NIRSDetectorsTable containing the
+# label and location of optical sources for the device
+detectors = NIRSDetectorsTable()
+# add a row for each detector
+detectors.add_row(label="D1", x=0.0, y=-1.0) # the z-coordinate is optional
+detectors.add_row(label="D2", x=0.0, y=1.0)
+
+
+# create a NIRSChannelsTable which defines the channels
+# between the provided sources and detectors
+channels = NIRSChannelsTable(sources, detectors)
+# each channel is composed of a single source, a single detector, and the wavelength
+# most source-detector pairs will use two separate wavelengths, and have two channels
+for wavelength in [690.0, 830.0]:
+    # for the source and detector parameters, pass in the index of
+    # the desired source (detector) in the sources (detectors) table
+    channels.add_row(label=f"S1.D1.{wavelength:.0f}nm", source=0,
+                     detector=0, source_wavelength=wavelength)
+    channels.add_row(label=f"S1.D2.{wavelength:.0f}nm", source=0,
+                     detector=1, source_wavelength=wavelength)
+    channels.add_row(label=f"S2.D1.{wavelength:.0f}nm", source=1,
+                     detector=0, source_wavelength=wavelength)
+    channels.add_row(label=f"S2.D2.{wavelength:.0f}nm", source=1,
+                     detector=1, source_wavelength=wavelength)
```

Comment:
`source_wavelength` will be correct but on this branch we've only defined the `wavelength` parameter (we haven't  updated `wavelength` -> `source_wavelength` and `emission_wavelength` on this branch) Should we merge  `enh/add-additional-snirf-fields` into `enh/add-documentation` (or equivalent merge `main` here once we merge PR #3)?

Reply:
Just merged in main after accepting #3, so we should be all set.